distributed/client.py

```diff
@@ -1308,7 +1301,11 @@ def _handle_error(self, exception=None):
 
             if not fast:
                 with ignoring(TimeoutError):
-                    await gen.with_timeout(timedelta(seconds=2), list(coroutines))
+
+                    async def _(c):
+                        await asyncio.wait_for(c, 2)
+
+                    await asyncio.gather(*[_(c) for c in coroutines])
```

Comment:
I'm curious, why the helper function here?

Reply:
Why not

```python
await asyncio.wait_for(asyncio.gather(*coroutines), 2)
```

This blocks until:
- all coroutines complete
- or 2 seconds passes

On timeout, all coroutines in `coroutines` are cancelled and a `TimeoutError` is raised.

Reply:
Ah, yes. That's much cleaner. Thanks for the suggestion @jcrist!